src/kitovu/sync/syncing.py

```diff
@@ -51,17 +56,51 @@ def start(connection_settings: ConnectionSettings) -> None:
             # each plugin should now yield all files recursively with list_path
             print(f'Downloading: {item}')
 
-            digest = plugin.create_remote_digest(item)
-            print(f'Remote digest: {digest}')
-
-            output = pathlib.Path(local_path / item.relative_to(remote_path))
-
-            pathlib.Path(output.parent).mkdir(parents=True, exist_ok=True)
-
-            with output.open('wb') as fileobj:
-                plugin.retrieve_file(item, fileobj)
-
-            digest = plugin.create_local_digest(output)
-            print(f'Local digest: {digest}')
-
+            remote_digest = plugin.create_remote_digest(item)
+            print(f'Remote digest: {remote_digest}')
+
+    # special filecache cases which FIXME here
+    # 1. remote deleted (triggers exception), local exists => REMOTE*
+    # 2. remote deleted (triggers exception), local exists AND changed (local_digest and cached_digest differ) => BOTH*
+
+            # if file doesn't exist: case 3 (normal case)
+            if not pathlib.Path(local_path / item.relative_to(remote_path)).exists():
```

Comment:
There's a lot of duplication with code handling `Filestate.REMOTE` here, which could be simplified. Couldn't the cache check whether the local file exists and return `Filestate.REMOTE` if it doesn't?